tools/workspace-plugin/src/executors/build/executor.spec.ts

```diff
@@ -1,18 +1,166 @@
-import { ExecutorContext } from '@nx/devkit';
+import { type ExecutorContext, logger, stripIndents } from '@nx/devkit';
 
 import { BuildExecutorSchema } from './schema';
 import executor from './executor';
+import { join } from 'node:path';
+import { existsSync, readFileSync, readdirSync } from 'node:fs';
+
+// ===== mocks start =====
+import { rm } from 'node:fs/promises';
+// ===== mocks end =====
+
+const options: BuildExecutorSchema = {
+  sourceRoot: 'src',
+  outputPathRoot: 'libs/proj/dist',
+  moduleOutput: [
+    {
+      module: 'commonjs',
+      outputPath: 'lib-commonjs',
+    },
+    {
+      module: 'es6',
+      outputPath: 'lib',
+    },
+  ],
+  assets: [
+    {
+      input: 'libs/proj/',
+      output: 'assets/',
+      glob: '*.md',
+    },
+  ],
+  clean: true,
+};
+
+const workspaceRoot = join(__dirname, '__fixtures__/executor');
 
-const options: BuildExecutorSchema = {};
 const context: ExecutorContext = {
-  root: '',
+  root: workspaceRoot,
   cwd: process.cwd(),
   isVerbose: false,
+  projectName: 'proj',
+  projectsConfigurations: {
+    version: 2,
+    projects: {
+      proj: {
+        root: 'libs/proj',
+        name: 'proj',
+      },
+    },
+  },
 };
 
+jest.mock('node:fs/promises', () => {
+  const actualFs = jest.requireActual('node:fs/promises');
+
+  return {
+    ...actualFs,
+    rm: jest.fn(actualFs.rm),
+  };
+});
+
+const rmMock = rm as jest.Mock;
+
 describe('Build Executor', () => {
```

Comment:
note: this is an e2e test - most crucial part of our toolchain needs to have e2e coverage